FirebaseAdmin/FirebaseAdmin/Messaging/FirebaseMessaging.cs

```diff
@@ -0,0 +1,171 @@
+        }
+
+        /// <summary>
+        /// Sends a message to the FCM service for delivery. The message gets validated both by
+        /// the Admin SDK, and the remote FCM service. A successful return value indicates
+        /// that the message has been successfully sent to FCM, where it has been accepted by the
+        /// FCM service.
+        /// <para>If the <paramref name="dryRun"/> option is set to true, the message will not be
+        /// actually sent to the recipients. Instead, the FCM service performs all the necessary
+        /// validations, and emulates the send operation. This is a good way to check if a
+        /// certain message will be accepted by FCM for delivery.</para>
+        /// </summary>
+        /// <returns>A task that completes with a message ID string, which represents
+        /// successful handoff to FCM.</returns>
+        /// <exception cref="ArgumentNullException">If the message argument is null.</exception>
+        /// <exception cref="ArgumentException">If the message contains any invalid
+        /// fields.</exception>
+        /// <exception cref="FirebaseException">If an error occurs while sending the
+        /// message.</exception>
+        /// <param name="message">The message to be sent. Must not be null.</param>
+        /// <param name="dryRun">A boolean indicating whether to perform a dry run (validation
+        /// only) of the send.</param>
+        public async Task<string> SendAsync(Message message, bool dryRun)
+        {
+            return await SendAsync(message, dryRun, default(CancellationToken));
+        }
+
+        /// <summary>
+        /// Sends a message to the FCM service for delivery. The message gets validated both by
+        /// the Admin SDK, and the remote FCM service. A successful return value indicates
+        /// that the message has been successfully sent to FCM, where it has been accepted by the
+        /// FCM service.
+        /// <para>If the <paramref name="dryRun"/> option is set to true, the message will not be
+        /// actually sent to the recipients. Instead, the FCM service performs all the necessary
+        /// validations, and emulates the send operation. This is a good way to check if a
+        /// certain message will be accepted by FCM for delivery.</para>
+        /// </summary>
+        /// <returns>A task that completes with a message ID string, which represents
+        /// successful handoff to FCM.</returns>
+        /// <exception cref="ArgumentNullException">If the message argument is null.</exception>
+        /// <exception cref="ArgumentException">If the message contains any invalid
+        /// fields.</exception>
+        /// <exception cref="FirebaseException">If an error occurs while sending the
+        /// message.</exception>
+        /// <param name="message">The message to be sent. Must not be null.</param>
+        /// <param name="dryRun">A boolean indicating whether to perform a dry run (validation
```

Comment:
I'm not sure if it's worth changing, but I don't think our documentation of `dryRun` is really sufficient. The client isn't doing a dry-run. I mean, it's actually sending the request to the backend. It's the backend that's doing a dry-run, not sending notifications back to clients. Maybe that's already clear to users of this API, but it surprised me.

Reply:
It is explained a bit in the `<summary>` section. I've also added a bit more details to the `<param>` section.